src/python_testing/TC_ACL_2_11.py

```diff
@@ -160,12 +193,18 @@ async def test_TC_ACL_2_11(self):
             server_list = ret[endpoint][Clusters.Descriptor][Clusters.Descriptor.Attributes.ServerList]
             for server in server_list:
                 cluster = Clusters.ClusterObjects.ALL_CLUSTERS[server]
-                await dev_ctrl.ReadAttribute(dut_node_id, [endpoint,
-                                                           cluster.Attributes.GeneratedCommandList,
-                                                           cluster.Attributes.AcceptedCommandList,
-                                                           cluster.Attributes.AttributeList,
-                                                           cluster.Attributes.FeatureMap,
-                                                           cluster.Attributes.ClusterRevision])
+                data = await dev_ctrl.ReadAttribute(dut_node_id, [(endpoint, cluster.Attributes.GeneratedCommandList),
+                                                                  (endpoint, cluster.Attributes.AcceptedCommandList),
+                                                                  (endpoint, cluster.Attributes.AttributeList),
+                                                                  (endpoint, cluster.Attributes.FeatureMap),
+                                                                  (endpoint, cluster.Attributes.ClusterRevision)])
+                for endpoint, clusters in data.items():
+                    for cluster, attributes in clusters.items():
+                        for attribute, value in attributes.items():
+                            asserts.assert_false(isinstance(value, ValueDecodeFailure) and
+                                                 isinstance(value.Reason, InteractionModelError) and
+                                                 value.Reason.status == Status.AccessRestricted,
```

Comment:
Wouldn't it be simpler to just assert that the status is success, period?

Reply:
When it works, the returned dictionary has valid values for each attribute of various types.  Not sure how to check for success.  The failure path for restrictions is a ValueDecodeFailure type containing an InteractionModelError which holds the AccessRestricted status.  Please let me know if you have a better method.

Reply:
I am not sure how to "check for success" in the Python stuff either, sorry, but there _has_ to be a way to check that the value is just a "server returned success"....  Please check with someone actually familiar with the Python bits?

Anyway, this can be a followup.

Reply:
Success of a read is having a value rather than a type indicating a failure. It's runtime typed. There is no easy way to get "success" as a thing, because success is no error

Reply:
Hmm, and there are path-specific failures other than ValueDecodeFailure and no common "these are all failures" super-class?

Reply:
Yes, because it's the attribute cache that is abstracted

Reply:
@bzbarsky-apple can we resolve this now?

Reply:
Yes, though it might be worth a comment in the code about how there is no sane way to check for "this read succeeded", and perhaps a followup issue about adding such a way.

Reply:
I added follow up to put a comment in this test file here: https://github.com/project-chip/connectedhomeip/issues/35341

I created a new issue for finding a better way to do this: https://github.com/project-chip/connectedhomeip/issues/35372

Reply:
Would success/failure not be `assert_false(isinstance(value, Error))` so that any error is bad? why do we filter it so narrowly? This requires a comment of the form: `only AccessRestricted is a failure here, because other failures such as ... or even IM errors such as ... are normal because ....`

Reply:
@andy31415 I captured your comment in the follow-up issue https://github.com/project-chip/connectedhomeip/issues/35341